src/java.base/share/classes/jdk/internal/loader/URLClassPath.java

```diff
@@ -569,11 +570,27 @@ private synchronized Loader getLoader(int index) {
             // Finally, add the Loader to the search path.
             loaders.add(loader);
             lmap.put(urlNoFragString, loader);
-           if (usingSharedClasses()) {                                          //OpenJ9-shared_classes_misc
-               /* update search path URLs */                                    //OpenJ9-shared_classes_misc
-               loaderURLs.add(url);                                             //OpenJ9-shared_classes_misc
-           }                                                                    //OpenJ9-shared_classes_misc
-                                                                                //OpenJ9-shared_classes_misc
+
+            if (usingSharedClasses()) {                                                  //OpenJ9-shared_classes_misc
+                /* update search path URLs */                                            //OpenJ9-shared_classes_misc
+                loaderURLs.add(url);                                                     //OpenJ9-shared_classes_misc
+            } else {                                                                     //OpenJ9-shared_classes_misc
+                if (!VM.isModuleSystemInited()) {                                        //OpenJ9-shared_classes_misc
+                    /**                                                                  //OpenJ9-shared_classes_misc
+                     * usingSharedClasses() is false. If Module System is not            //OpenJ9-shared_classes_misc
+                     * initialized yet, it is likely that BuiltinClassLoader             //OpenJ9-shared_classes_misc
+                     * hasn't called setSharedClassProvider(). We still nedd to          //OpenJ9-shared_classes_misc
+                     * update loaderURLs in this case.                                   //OpenJ9-shared_classes_misc
+                     */                                                                  //OpenJ9-shared_classes_misc
+                    if (null != loaderURLs) {                                            //OpenJ9-shared_classes_misc
+                        /* update search path URLs */                                    //OpenJ9-shared_classes_misc
+                        loaderURLs.add(url);                                             //OpenJ9-shared_classes_misc
```

Comment:
This is common code, the following can avoid duplicating it.
```
if (null == loaderURLs) { 
    loaderURLs = new ArrayList(path.size());
}
loaderURLs.add(url);
```

Reply:
Fixed.